cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -69,22 +79,54 @@ def create_cluster(
     :param client_token: Idempotency token that can be set by the client so that retries for the same request are
     idempotent
     """
+    # Set defaults
+    rollback_on_failure = rollback_on_failure or True
+    validation_failure_level = validation_failure_level or ValidationLevel.ERROR
+    dryrun = dryrun or False
     create_cluster_request_content = CreateClusterRequestContent.from_dict(create_cluster_request_content)
-    if create_cluster_request_content.cluster_configuration == "invalid":
-        raise CreateClusterBadRequestException(
-            CreateClusterBadRequestExceptionResponseContent(configuration_validation_errors=[], message="invalid")
+
+    # Validate inputs
+    if client_token:
+        raise BadRequestException("clientToken is currently not supported for this operation")
+    cluster_config_dict = _parse_cluster_config(create_cluster_request_content.cluster_configuration)
+
+    # Check unique cluster name
+    cluster = Cluster(create_cluster_request_content.name, cluster_config_dict)
+    if AWSApi.instance().cfn.stack_exists(cluster.stack_name):
+        raise ConflictException(f"cluster {cluster.name} already exists")
+
+    # Create cluster
+    try:
+        stack_id, ignored_validation_failures = cluster.create(
+            disable_rollback=not rollback_on_failure,
+            validator_suppressors=_get_validator_suppressors(suppress_validators),
+            validation_failure_level=FailureLevel[validation_failure_level],
+            dryrun=dryrun,
         )
 
-    return CreateClusterResponseContent(
-        ClusterInfoSummary(
-            cluster_name="nameeee",
-            cloudformation_stack_status=CloudFormationStatus.CREATE_COMPLETE,
-            cloudformation_stack_arn="arn",
-            region="region",
-            version="3.0.0",
-            cluster_status=ClusterStatus.CREATE_COMPLETE,
+        if dryrun:
+            LOGGER.info("Skipping cluster creation due to dryrun operation")
+            raise DryrunOperationException()
+
+        return CreateClusterResponseContent(
+            ClusterInfoSummary(
+                cluster_name=create_cluster_request_content.name,
+                cloudformation_stack_status=CloudFormationStatus.CREATE_IN_PROGRESS,
+                cloudformation_stack_arn=stack_id,
+                region=os.environ.get("AWS_DEFAULT_REGION"),
+                version=get_installed_version(),
+                cluster_status=cloud_formation_status_to_cluster_status(CloudFormationStatus.CREATE_IN_PROGRESS),
+            ),
+            validation_messages=_build_validation_messages_list(ignored_validation_failures) or None,
+        )
+    except ConfigValidationError as e:
+        raise _handle_config_validation_error(e)
+    except ClusterActionError as e:
```

Comment:
Asking because this is also relevant to my work on the `DeleteImage`. The `Cluster` class wraps exceptions in a `ClusterActionError`, which is surfaced as a 500, however I note that this might also be caused by `AWSClientError`, for example if the `AWSApi.instance().cfn.create_stack_from_url` or `AWSApi.instance().cfn.describe_stack` have a `REQUEST_LIMIT_EXCEEDED`, which I think we would like to surface as a 429 `LimitExceededException`